src/main/java/com/limechain/trie/AccessorHolder.java

```diff
@@ -0,0 +1,27 @@
+package com.limechain.trie;
+
+import io.emeraldpay.polkaj.types.Hash256;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+public class AccessorHolder {
+    private BlockTrieAccessor blockTrieAccessor;
+
+    private static AccessorHolder instance;
+
+    private AccessorHolder() { }
+
+    public static AccessorHolder getInstance() {
+        if (instance == null) {
+            instance = new AccessorHolder();
+        }
+        return instance;
+    }
+
+    public BlockTrieAccessor setBlock(Hash256 blockHash) {
```

Comment:
I'd rename this method to `setAtBlock` (or similar) to indicate that aside from the returned instance is a mutated `this`... or add a Javadoc. (I fear side effects)